src/BlazorWebView/src/Maui/Windows/WinUIWebViewManager.cs

```diff
@@ -85,113 +84,86 @@ protected override async Task HandleWebResourceRequest(CoreWebView2WebResourceRe
-					stream = await CopyContentToRandomAccessStreamAsync(contentStream);
+					relativePath = _hostPageRelativePath;
 				}
-			}
-			else
-			{
-				var path = Path.Combine(AppContext.BaseDirectory, relativePath);
-				if (File.Exists(path))
+				relativePath = Path.Combine(_contentRootRelativeToAppRoot, relativePath.Replace('/', '\\'));
+				statusCode = 200;
+				statusMessage = "OK";
+				var contentType = StaticContentProvider.GetResponseContentTypeOrDefault(relativePath);
+				headers = StaticContentProvider.GetResponseHeaders(contentType);
+				IRandomAccessStream? stream = null;
+				if (_isPackagedApp)
 				{
-					using var contentStream = File.OpenRead(path);
-					stream = await CopyContentToRandomAccessStreamAsync(contentStream);
+					var winUIItem = await Package.Current.InstalledLocation.TryGetItemAsync(relativePath);
+					if (winUIItem != null)
+					{
+						using var contentStream = await Package.Current.InstalledLocation.OpenStreamForReadAsync(relativePath);
+						stream = await CopyContentToRandomAccessStreamAsync(contentStream);
+					}
+				}
+				else
+				{
+					var path = Path.Combine(AppContext.BaseDirectory, relativePath);
+					if (File.Exists(path))
+					{
+						using var contentStream = File.OpenRead(path);
+						stream = await CopyContentToRandomAccessStreamAsync(contentStream);
+					}
 				}
-			}
 
-			var hotReloadedContent = Stream.Null;
-			if (StaticContentHotReloadManager.TryReplaceResponseContent(_contentRootRelativeToAppRoot, requestUri, ref statusCode, ref hotReloadedContent, headers))
-			{
-				stream = await CopyContentToRandomAccessStreamAsync(hotReloadedContent);
-			}
+				var hotReloadedContent = Stream.Null;
+				if (StaticContentHotReloadManager.TryReplaceResponseContent(_contentRootRelativeToAppRoot, requestUri, ref statusCode, ref hotReloadedContent, headers))
+				{
+					stream = await CopyContentToRandomAccessStreamAsync(hotReloadedContent);
+				}
 
-			if (stream != null)
-			{
-				var headerString = GetHeaderString(headers);
+				if (stream != null)
+				{
+					var headerString = GetHeaderString(headers);
 
-				_logger.ResponseContentBeingSent(requestUri, statusCode);
+					_logger.ResponseContentBeingSent(requestUri, statusCode);
 
-				eventArgs.Response = _coreWebView2Environment!.CreateWebResourceResponse(
-					stream,
-					statusCode,
-					statusMessage,
-					headerString);
+					eventArgs.Response = _coreWebView2Environment!.CreateWebResourceResponse(
+						stream,
+						statusCode,
+						statusMessage,
+						headerString);
+				}
+				else
+				{
+					_logger.ReponseContentNotFound(requestUri);
+				}
 
-				return true;
-			}
-			else
-			{
-				_logger.ReponseContentNotFound(requestUri);
+				async Task<IRandomAccessStream> CopyContentToRandomAccessStreamAsync(Stream content)
+				{
+					using var memStream = new MemoryStream();
+					await content.CopyToAsync(memStream);
```

Comment:
That can't be the case. The failure is at build time, and that code only executes at runtime.

Reply:
Hum this is very strange your fixes something related with static file handling and the log says it fails on static web assets.. it very seems to be related. 

```
D:\a\_work\1\s\bin\dotnet\sdk\8.0.100\Sdks\Microsoft.NET.Sdk.StaticWebAssets\targets\Microsoft.NET.Sdk.StaticWebAssets.Publish.targets(25,5): error : IOException: The process cannot access the file 'D:\a\_work\1\s\src\Controls\samples\Controls.Sample\obj\Debug\net8.0-android\staticwebassets.publish.json' because it is being used by another process. [D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Maui.Controls.Sample.csproj::TargetFramework=net8.0-android]

Reply:
@rmarinho that file is not used at runtime.